runtime/integration-tests/src/generic/cases/liquidity_pools.rs

```diff
@@ -1780,23 +1842,368 @@ mod centrifuge {
 		}
 
 		fn restrict_lp_eth_usdc_transfer<T: Runtime>() {
-			todo!()
+			let mut env = RuntimeEnv::<T>::from_parachain_storage(
+				Genesis::default()
+					.add(genesis::balances::<T>(cfg(10)))
+					.add(orml_tokens::GenesisConfig::<T> {
+						balances: vec![(
+							Keyring::Alice.to_account_id(),
+							LP_ETH_USDC,
+							T::ExistentialDeposit::get() + lp_eth_usdc(TRANSFER_AMOUNT),
+						)],
+					})
+					.storage(),
+			);
+
+			env.parachain_state_mut(|| {
+				register_lp_eth_usdc::<T>();
+
+				let pre_transfer_alice = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Alice.to_account_id(),
+				);
+				let pre_transfer_bob = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Bob.to_account_id(),
+				);
+				let pre_transfer_charlie = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Charlie.to_account_id(),
+				);
+
+				add_allowance::<T>(
+					Keyring::Alice,
+					LP_ETH_USDC,
+					Location::Local(Keyring::Bob.to_account_id()),
+				);
+
+				assert_noop!(
+					pallet_restricted_tokens::Pallet::<T>::transfer(
+						RawOrigin::Signed(Keyring::Alice.into()).into(),
+						Keyring::Charlie.into(),
+						LP_ETH_USDC,
+						lp_eth_usdc(TRANSFER_AMOUNT)
+					),
+					pallet_restricted_tokens::Error::<T>::PreConditionsNotMet
+				);
+
+				let after_transfer_alice = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Alice.to_account_id(),
+				);
+				let after_transfer_charlie = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Charlie.to_account_id(),
+				);
+
+				assert_eq!(after_transfer_alice, pre_transfer_alice);
+				assert_eq!(after_transfer_charlie, pre_transfer_charlie);
+
+				assert_ok!(pallet_restricted_tokens::Pallet::<T>::transfer(
+					RawOrigin::Signed(Keyring::Alice.into()).into(),
+					Keyring::Bob.into(),
+					LP_ETH_USDC,
+					lp_eth_usdc(TRANSFER_AMOUNT)
+				),);
+
+				let after_transfer_alice = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Alice.to_account_id(),
+				);
+				let after_transfer_bob = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Bob.to_account_id(),
+				);
+				let after_transfer_charlie = orml_tokens::Pallet::<T>::free_balance(
+					LP_ETH_USDC,
+					&Keyring::Charlie.to_account_id(),
+				);
+
+				assert_eq!(
+					after_transfer_alice,
+					pre_transfer_alice - lp_eth_usdc(TRANSFER_AMOUNT)
+				);
+				assert_eq!(
+					after_transfer_bob,
+					pre_transfer_bob + lp_eth_usdc(TRANSFER_AMOUNT)
+				);
+				assert_eq!(after_transfer_charlie, pre_transfer_charlie);
+			});
 		}
 
-		fn restrict_lp_eth_usdc_xcm_transfer<T: Runtime>() {
-			todo!()
+		fn restrict_lp_eth_usdc_lp_transfer<T: Runtime + FudgeSupport>() {
```

Comment:
@mustermeiszer here's the test that you mentioned in a previous comment. I went the full way of adding a router just to confirm that the LP call goes through so that we can also check that the domain account gets the transfer amount. WDYT